File: backend/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  category: {
    type: String,
    enum: ['légumes', 'fruits', 'céréales', 'produits_laitiers', 'viandes'],
    required: true
  },
  price: { type: Number, required: true },
  unit: { type: String, enum: ['kg', 'unité', 'litre', 'pièce'], required: true },
  stock: { type: Number, required: true, default: 0 },
  imageUrl: { type: String, default: '' },
  description: { type: String, default: '' },
  farmerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Product', productSchema);

File: flutter_app/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String category;
  final double price;
  final String unit;
  final int stock;
  final String imageUrl;
  final String description;

  Product({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    required this.unit,
    required this.stock,
    this.imageUrl = '',
    this.description = '',
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      category: json['category'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      unit: json['unit'] ?? 'kg',
      stock: json['stock'] ?? 0,
      imageUrl: json['imageUrl'] ?? '',
      description: json['description'] ?? '',
    );
  }
}
